### PR TITLE
flake8 contrib/ and lib/tool_shed/ .

### PR DESCRIPTION
**Pull request by nsoranzo**
Also:
- use `sys.exit()` instead of `exit()`
- fix `sys.path` manipulations.

### CODE REVIEW

File: lib/tool_shed/scripts/api/common.py [hunk 1]
```diff
@@ -4,12 +4,9 @@
 import urllib
 import urllib2
 
-new_path = [ os.path.join( os.path.dirname( __file__ ), os.pardir, os.pardir, os.pardir, os.pardir, 'lib' ) ]
-new_path.extend( sys.path[ 1: ] )
-sys.path = new_path
+sys.path.insert(1, os.path.join( os.path.dirname( __file__ ), os.pardir, os.pardir, os.pardir ) )
```
Comment (jmchilton) [inline lib/tool_shed/scripts/api/common.py:7]: This doesn't have to end with `lib`?

**Comment by jmchilton** [comment]
:+1:

File: lib/tool_shed/scripts/api/common.py (hunk 1 above)
Comment (nsoranzo) [inline lib/tool_shed/scripts/api/common.py:7]: No, this file is already in a subdirectory of `lib`, so I have removed `os.pardir, 'lib'` .

**Comment by martenson** [comment]
This breaks 2 tests for me (`./run_tests.sh -toolshed`): 

```
======================================================================
ERROR: Verify the exported capsule contents.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/marten/devel/git/galaxy/test/tool_shed/functional/test_0490_export_import_repositories.py", line 82, in test_0015_verify_exported_capsule
    self.verify_capsule_contents( capsule_filepath, owner=common.test_user_1_name )
  File "/Users/marten/devel/git/galaxy/test/tool_shed/base/twilltestcase.py", line 1313, in verify_capsule_contents
    tar_object = tarfile.open( capsule_filepath, 'r:*' )
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/tarfile.py", line 1651, in open
    raise ValueError("nothing to open")
ValueError: nothing to open

======================================================================
ERROR: Import the exported repository capsule.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/marten/devel/git/galaxy/test/tool_shed/functional/test_0490_export_import_repositories.py", line 94, in test_0020_import_repository_capsule
    strings_displayed_after_submit=[ 'Import not necessary', 'Exists' ] )
  File "/Users/marten/devel/git/galaxy/test/tool_shed/base/twilltestcase.py", line 829, in import_capsule
    tc.formfile( 'upload_capsule', 'file_data', filename )
  File "/Users/marten/devel/git/galaxy/eggs/twill-0.9-py2.7.egg/twill/commands.py", line 481, in formfile
    fp = open(filename, 'rb')
IOError: [Errno 2] No such file or directory: ''
```

**Comment by martenson** [comment]
seems the tests were failing randomly, :+1:

**Comment by nsoranzo** [comment]
xref: #391

**Pull request merged**